networkx/algorithms/reciprocity.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+#
+#    Copyright (C) 2015 by 
+#    Haochen Wu <wuhaochen42@gmail.com>
+#    All rights reserved.
+#    BSD license.
+"""Algorithms to calculate reciprocity in a directed graph."""
+from networkx import NetworkXError
+from ..utils import not_implemented_for
+
+__author__ = """\n""".join(['Haochen Wu <wuhaochen42@gmail.com>'])
+
+__all__= ['reciprocity','overall_reciprocity']
+
+@not_implemented_for('undirected','multigraph')
+def reciprocity(G, nodes=None):
+    """Computer the reciprocity in a directed graph.
```

Comment:
Thanks for all the suggestions. I'll work on them tomorrow.
Is there a recommended way to give warnings when it's not defined?